### PR TITLE
Add optional `tag` config for better log processing

### PR DESCRIPTION
**Pull request by Violet-E**
different tag for different hosts helps processing logs better

### CODE REVIEW

File: coraza.go [hunk 1]
```diff
@@ -30,9 +31,9 @@ type corazaModule struct {
 	// deprecated
 	Include    []string `json:"include"`
 	Directives string   `json:"directives"`
-
-	logger *zap.Logger
-	waf    coraza.WAF
+	Tag        string   `json:"tag"`
```
Comment (jcchavezs) [inline coraza.go:34]: I wonder if tag is the right name. Is this something idiomatic in caddy?

File: coraza.go [hunk 2]
```diff
@@ -182,23 +186,29 @@ func parseCaddyfile(h httpcaddyfile.Helper) (caddyhttp.MiddlewareHandler, error)
...
 
-func newErrorCb(logger *zap.Logger) func(types.MatchedRule) {
+func newErrorCb(logger *zap.Logger, tag string) func(types.MatchedRule) {
 	return func(mr types.MatchedRule) {
 		data := mr.ErrorLog(403)
+		var msg string
+		if tag == "" {
+			msg = data
+		} else {
+			msg = fmt.Sprintf("[tag %s] %s", tag, data)
```
Comment (jcchavezs) [inline coraza.go:196]: The tag prefix interpolation might happen only oncean be referenced in this callback I think.

File: coraza.go (hunk 1 above)
Comment (Violet-E) [reply coraza.go:34]: I use it to distinguish different vhosts, such as the domain  of the reverse proxy in the current site，what's your advice

File: coraza.go (hunk 2 above)
Comment (Violet-E) [reply coraza.go:196]: this tag will be happen every time it breaks the rule,I think it's enough for debug

File: coraza.go (hunk 1 above)
Comment (jcchavezs) [reply coraza.go:34]: I think something like detaulfLabels or defaultTags being a key value would work.

**Comment by Violet-E** [comment]
I don't know if it's necessary for other people. I'm trying to build waf
for multiuser,but found I have noway to figure out which vhost the log come
from

José Carlos Chávez ***@***.***> 于 2023年5月8日周一 下午1:26写道：

**Pull request closed without merging**